lib/+ne7/+ui/paintMasks.m

```diff
@@ -0,0 +1,328 @@
+function masks = paintMasks(im)
+% lets the user outline masks in the image on the current axis
+
+% process image for assisted segmentation
+processed_image = (imfilter(imfill(im),gausswin(2)*gausswin(2)'));
+
+% initialize parameters
+radius = 10; % mask pixel radius
+sz = size(im);
+im = (im - min(im(:)))./(max(im(:))-min(im(:))); % normalize image
+masks = zeros(sz);
+colors = 0;
+sat = 0.5;
+sat_tog = 0.5;
+undoBuffer = {masks};
+winsz = 32; % max size of the pointer window
+running = true;
+drawing = false;
+neuron = 0;
+xLoc = 0;
+yLoc = 0;
+assisted = false;
+thresh = 0.01;
+fit_center = false(winsz,winsz);fit_center(round(winsz/2),round(winsz/2))=true;
+fit_mask = zeros(winsz,winsz);
+ppitch = 1;
+hp =[];
+
+% Plot
+hf = figure('NumberTitle','off',...
+    'Name','Paint masks',...
+    'KeyPressFcn',@dispkeyevent,...
+    'WindowScrollWheelFcn', @adjMaskSize,...
+    'HitTest','off');
+h = image(im);
+axis image
+set(gca,'xtick',[],'ytick',[])
+set(h,'buttondownfcn',@updateMasks)
+printInstructions
+redraw
+adjMaskSize
+hold on
+
+% wait until done
+while running && nargout>0
+    try if ~ishandle(h);break;end;catch;break;end
+    pause(0.1);
+end
+
+function dispkeyevent(~, event)
+    switch event.Key
+        case 'a' % assisted segmentation
+            if assisted
+                assisted = false;
+                try delete(hp);end
+                set(hf,'WindowButtonMotionFcn','','WindowButtonUpFcn','');
+                set(hf,'WindowScrollWheelFcn', @adjMaskSize)
+                set(gcf,'pointer','custom','PointerShapeHotSpot',[winsz winsz]/2)
+                adjMaskSize
+            else
+                c = nan(16,16);
+                c(8,4:12)=2;
+                c(4:12,8)=2;
+                assisted = true;
+                adjMaskFit
+                set(hf,'WindowScrollWheelFcn', @adjMaskFit)
+                set(hf,'WindowButtonMotionFcn',@adjMaskFit)
+                set(gcf,'PointerShapeCData',c,'pointer','custom','PointerShapeHotSpot',[8 8])
+            end
+        case 'backspace' % UNDO
+            if length(undoBuffer)>1
+                masks = undoBuffer{end-1};
+                undoBuffer(end-1:end) = [];
+                redraw
+            end
+        case 'space'  % SPACE - toggle outlines
+            if sat>0
+                sat_tog = sat;
+                sat = 0;
+                redraw
+            else
+                sat = sat_tog;
+                redraw
+            end
+        case 'leftbracket'
+            if sat>=0.1
+                sat = sat-0.1;
```

Comment:
It is clamped to [0 1] . From 0 it can only increase and 1 it can only decrease.